sentinel.go

```diff
@@ -433,10 +437,26 @@ func (c *sentinelFailover) closeSentinel() error {
 }
 
 func (c *sentinelFailover) RandomSlaveAddr(ctx context.Context) (string, error) {
-	addresses, err := c.slaveAddrs(ctx)
-	if err != nil {
-		return "", err
+	if c.opt == nil {
+		return "", errors.New("opt is nil")
 	}
+
+	var addresses []string
+	var err error
+
+	for _, allowDisconnected := range []bool{false, true} {
```

Comment:
Can we simplify this? It is getting late here and I am having troubles trying to understand what this does...

If it is not possible - please add a comment explaining what is happening.